test/dox/doc_test.py

```diff
@@ -0,0 +1,32 @@
+import subprocess
+import sys
+import re
+
+def getopts(argv):
+    opts = {}  # Empty dictionary to store key-value pairs.
+    while argv:  # While there are arguments left to parse...
+        if argv[0][0] == '-':  # Found a "-name value" pair.
+            opts[argv[0]] = argv[1]  # Add key and value to the dictionary.
+        argv = argv[1:]  # Reduce the argument list by copying it starting from index 1.
+    return opts
+
+if __name__ == '__main__':
+    from sys import argv
+    myargs = getopts(argv)
+
+    '''Use regex to filter CLANG option warnings'''
+    regexp = re.compile(r'(CLANG_OPTIONS|CLANG_ASSISTED_PARSING)')
+
+    process = subprocess.Popen([myargs.get("-e"), myargs.get("-i")], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    cnt = 0
+    for line in process.stdout:
+        _line = line.decode('utf-8')
+        if "warning:" in _line and not regexp.search(_line):
+            cnt = cnt + 1
+            print(line)
+
+    if cnt != 0:
+        print("Detected %d warnings!" % (cnt))
+        sys.exit(1)
+
+    sys.exit(0)
```

Comment:
why do we need this python script? I would really like to keep anything python out of the repo. Why can we not just grep or if need be filter with perl?

Reply:
I just copied the files from the infrastructure repo.

Reply:
That's not an excuse :wink: 

No, seriously, this file can be replaced by one line of grep I think.

Reply:
Can we please merge this, I don't want to waste time to get this working with other tools. This should have been a quick pr, but this (again) needed 5 days of discussion, because of no real reasons. This is really not good for any flow. Can you please accept for once a working solution?

We can open a new issue if you want, but I personally don't see a reason against a multi-platform working solution. grep and bash is limited to linux, we might hack some cmake scripts to get the same semantics. But, this will all take time, which is out of scope of a quick fix to get the documentation test up again.

@rrahn any thoughts?